pandas/core/frame.py

```diff
@@ -3599,7 +3599,7 @@ def align(
         see_also_sub=" or columns",
     )
     @Appender(NDFrame.set_axis.__doc__)
-    def set_axis(self, labels, axis=0, inplace=False):
+    def set_axis(self, labels, axis: Axis = 0, inplace: bool = False):
```

Comment:
return type and labels?

Reply:
return type is easy.  for labels I'm ambivalent since it _could_ be listlike, but _should_ be Index

Reply:
i take it back, return type depends on inplace

Reply:
> i take it back, return type depends on inplace

yeah, that is causing us problems without Literal.